backend/pkg/database/gorm_common.go

```diff
@@ -179,6 +180,12 @@ func (gr *GormRepository) DeleteCurrentUser(ctx context.Context) error {
 	return nil
 }
 
+func (gr *GormRepository) GetUsers(ctx context.Context) ([]models.User, error) {
+	var users []models.User
+	result := gr.GormClient.WithContext(ctx).Find(&users)
+	return users, result.Error
```

Comment:
as a security step, I would sanitize the user's passwords in this layer. I know you do it elsewhere, but I think it makes sense to move that code here.